keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/ApplicationControllerTests.kt

```diff
@@ -171,17 +199,38 @@ internal class ApplicationControllerTests : JUnit5Minutests {
         deliveryConfig.environments.flatMap { it.resources }.forEach { resource ->
           resourceRepository.appendHistory(ResourceValid(resource))
         }
-        artifactRepository.store(artifact, "fnord-1.0.0-h0.2add2c9", ArtifactStatus.RELEASE)
-        artifactRepository.store(artifact, "fnord-1.0.1-h1.2add2c9", ArtifactStatus.RELEASE)
-        artifactRepository.store(artifact, "fnord-1.0.2-h2.2add2c9", ArtifactStatus.RELEASE)
-        artifactRepository.store(artifact, "fnord-1.0.3-h3.2add2c9", ArtifactStatus.RELEASE)
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.0-h0.2add2c9", "test")
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.0-h0.2add2c9", "staging")
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.0-h0.2add2c9", "production")
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.1-h1.2add2c9", "test")
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.1-h1.2add2c9", "staging")
-        artifactRepository.markAsSuccessfullyDeployedTo(deliveryConfig, artifact, "fnord-1.0.2-h2.2add2c9", "test")
-        artifactRepository.approveVersionFor(deliveryConfig, artifact, "fnord-1.0.3-h3.2add2c9", "test")
+        artifactRepository.store(artifact, "fnord-1.0.0-h0.a0a0a0a", ArtifactStatus.RELEASE)
+        artifactRepository.store(artifact, "fnord-1.0.1-h1.b1b1b1b", ArtifactStatus.RELEASE)
+        artifactRepository.store(artifact, "fnord-1.0.2-h2.c2c2c2c", ArtifactStatus.RELEASE)
+        artifactRepository.store(artifact, "fnord-1.0.3-h3.d3d3d3d", ArtifactStatus.RELEASE)
```

Comment:
@emjburns I took the trouble of using different hashes for each version. 🎁 😄

Reply:
wowie! So realistic 😍